docs/proposals/20191017-kubeadm-based-control-plane.md

```diff
@@ -173,12 +179,47 @@ With the following validations:
 - `KubeadmControlPlane.Spec.KubeadmConfigSpec` allows mutations required for supporting following use cases:
     - Change of imagesRepository/imageTags (with validation of CoreDNS supported upgrade)
     - Change of node registration options
-    - Change of pre/post kubeadm commands 
+    - Change of pre/post kubeadm commands
     - Change of cloud init files
 
 And the following defaulting:
 
-- `KubeadmControlPlane.Spec.Replicas: 1` 
+- `KubeadmControlPlane.Spec.Replicas: 1`
+
+##### Changes for supporting different upgrade strategies
+
+```go
+type RolloutType string
+
+const (
+  // ScaleDownRolloutType constant is to set rollout strategy type to scale down.
+  ScaleDownRolloutType  RolloutType = "ScaleDown"  
+
+  // ScaleUpRolloutType constant is to set rollout strategy type to scale up.
+  ScaleUpRolloutType  RolloutType = "ScaleUp"
+)
+```
+- Add `KubeadmControlPlane.Spec.RolloutStrategy` defined as:
+
+```go
+    // RolloutStrategy is a field to set an kubeadm controlplane rollout strategy.
```

Comment:
The description should provide a bit more context: what's supported? what are the advantages between the strategies?